dist/tools/sliptty/start_network

```diff
@@ -0,0 +1,104 @@
+#! /bin/sh
+
+SLIPPY_DIR="$(cd "$(dirname "$0")" && pwd -P)"
+UHCPD="$(cd "${SLIPPY_DIR}/../uhcpd/bin" && pwd -P)/uhcpd"
+TUN=sl0
+UHCPD_PID=
+CREATED_IFACE=0
+
+unsupported_plattform() {
+    echo "unsupported platform" >&2
+    echo "(currently supported \`uname -s\` 'Darvin' and 'Linux')" >&2
+}
+
+case "$(uname -s)" in
+    Darvin)
+        PLATFORM="OSX";;
+    Linux)
+        PLATFORM="Linux";;
+    *)
+        unsupported_platform
+        exit 1
+        ;;
+esac
+
+create_tun() {
+    case "${PLATFORM}" in
+        Linux)
+            if ! ip link show ${TUN} 2>&1 > /dev/null; then
+                sudo ip tuntap add ${TUN} mode tun user ${USER} || exit 1
+                CREATED_IFACE=1
+            fi
+            sudo sysctl -w net.ipv6.conf.${TUN}.forwarding=1
+            sudo sysctl -w net.ipv6.conf.${TUN}.accept_ra=1
+            sudo ip link set ${TUN} up || exit 1
+            sudo ip address add fe80::1/64 dev ${TUN}
+            sudo ip neigh add fe80::2 dev ${TUN}
+            sudo ip route add ${PREFIX} via fe80::2 dev ${TUN}
+            ;;
+        OSX)
+            # TUN interface in OSX needs to be called tunX
+            TUN="tun$(echo "${TUN}" | grep -o "[0-9]\+$")"
+            sudo chown ${USER} /dev/${TUN} || exit 1
+            sudo sysctl -w net.inet6.ip6.forwarding=1
+            echo "start RIOT instance for ${TUN} now and hit enter"
+            read
+            sudo ifconfig ${TUN} up || exit 1
+            sudo ifconfig ${TUN} inet6 fe80::1 prefixlen 64
+            sudo route -n add -interface ${TUN} -inet6 -prefixlen 64 \
+                    ${PREFIX} fe80::2 || exit 1
+            ;;
+    esac
+    return 0
+}
+
+remove_tun() {
+    case "${PLATFORM}" in
+        Linux)
+            sudo ip tuntap del ${TUN} mode tun
+            ;;
+        OSX)
+            sudo route delete -inet6 ${PREFIX} -prefixlen 64 fe80::2
+            sudo sysctl -w net.inet6.ip6.forwarding=0
+            ;;
+    esac
+}
+
+cleanup() {
+    if [ 1 -eq "${CREATED_IFACE}" ]; then
+        remove_tun
+    fi
+    if [ -n "${UHCPD_PID}" ]; then
+        sudo kill ${UHCPD_PID}
+    fi
+    trap "" INT QUIT TERM EXIT
+}
+
+start_uhcpd() {
+    sudo ${UHCPD} ${TUN} ${PREFIX} > /dev/null &
+    UHCPD_PID=$!
+}
+
+usage() {
+    echo "usage: $1 [-I <sl0>] <prefix> serial [baudrate]"
+    echo "usage: $1 [-I <sl0>] <prefix> tcp:host [port]"
+}
+
+trap "cleanup" INT QUIT TERM EXIT
+
+while getopts hI: opt; do
+    case ${opt} in
+        I)  TUN=${OPTARG}; shift 2;;
+        h)  usage $0; exit 0;;
+    esac
+done
+
+PREFIX="$1"
+shift
+
+if [ -z "$PREFIX" ]; then
+    usage $0
+    exit 1
+fi
+
+create_tun && start_uhcpd && sudo ${SLIPPY_DIR}/sliptty -I ${TUN} "$*"
```

Comment:
why `"$*"` and not `"$@"`?
`"$*"` creates one argument, `"$@"` keeps them as separate arguments
short summary with example: https://stackoverflow.com/questions/3008695/what-is-the-difference-between-and-in-bash

Reply:
Did not know that.